### ISSUE TITLE
Can keras model run on specific device?

### ISSUE THREAD

**Issue by pswpswpsw** [fchollet/keras#4613]
## Problem configuration
- using tensorflow as backed. 
- computer with 1GPU card and 12 CPUs
- not distributed learning over cluster
- with only one session, use GPU or use CPUs. Not using both of them at any time.

###  run keras in CPU
`with tf.device(/cpu:0)`

### run keras in GPU
I don't write anything, let everything be as default. So since I install tensorflow in GPU version. It should assume default using GPU.
`with tf.device(/gpu:0)`

This undocumented trick works for me so far. However, since keras is a blackbox to me, while tensorflow is more structured and clear, I feel there should be an improvement for keras to better control the CPU/GPU device **with in keras**. I know we could just use keras as simplified tensorflow layer constructor. Thus it is possible to run everything under framework of tensorflow rather than living in the world of keras.

## Comment:
If you have alternative ways to force keras be used in CPU or GPU, please comment below and let everyone else know.

Best,
Shaowu

**Comment by tpinetz** [comment]
You can also set the environment variable: CUDA_VISIBLE_DEVICES to limit the number of gpus used. http://www.acceleware.com/blog/cudavisibledevices-masking-gpus

**Comment by MartinThoma** [comment]
Put

```
import os
os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"   # see issue #152
os.environ["CUDA_VISIBLE_DEVICES"] = ""
```

before tensorflow is imported.

**Comment by wajahat57** [comment]
import os
os.environ["CUDA_VISIBLE_DEVICES"] = '-1'             works on windows 10

**Comment by ThierryBachmann** [comment]
work perfect for me (windows 10 and keras 2.2.4)
Thank !

**Comment by lauraset** [comment]
> work perfect for me (windows 10 and keras 2.2.4)
> Thank !


when I did this in keras (with the backend of tensorflow), the program will occupy all the memories of GPU, but it only runs on one GPU. Anyone has meet this problem before?
